site/en/blog/has-with-cq-m105/index.md

```diff
@@ -0,0 +1,126 @@
+
+Container queries enable developers to query a parent selector for its size and styling information, making it possible for a child element to own its responsive styling logic, no matter where it lives on a web page.
+
+-- IMG TODO --
+
+Instead of relying on the viewport for styling input such as available space, developers now have the ability to query the size of in-page elements too.. This capability means that a component owns its responsive styling logic. This makes the component much more resilient, as the styling logic is attached to it, no matter where it appears on the page.
+
+{% Aside 'caution' %}
+Container queries are [currently supported](https://caniuse.com/css-container-queries) in Chromium and Safari TP.
+{% endAside %}
+
+## Using container queries
+
+To build with container queries, you must first set containment on a parent element. Do this by setting a `container-type` on the parent container. You might have a card with an image and some text content that looks like this:
+
+{% Img src="image/HodOHWjMnbNw56hvNASHWSgZyAf2/xR7RmgS1TUZmZ9ntOFrV.png", alt="single two-column card", width="795", height="182" %}
+
+To create a container query, set `container-type` on the card container:
+
+```css
+.card-container {
+  container-type: inline-size;
+}
+```
+
+Setting the `container-type` to `inline-size` says we want to query for the inline-direction size of the parent. In latin languages like english, this would be the width of the card, since the text flows inline from left to right.
+
+Now, we can use that container to apply styles to any of its children using `@container`:
+
+```css
+.card {
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+}
+
+@container (max-width: 400px) {
+  .card {
+    grid-template-columns: 1fr;
+  }
+}
+```
+
+{% Codepen {
+  user: 'una',
+  id: 'wvyXQXY',
+  height: 450,
+  tab: 'css, result'
+} %}
+
+## The :has() parent selector
+
+The CSS `:has()` pseudo-class enables developers to check if a parent element contains children with specific parameters. 
+
+For example, `p:has(span)` indicates a paragraph (`p`) selector, which has a `span` inside of it. You can use this to style the parent paragraph itself, *or* style anything within it. One useful example is `figure:has(figcaption)` to style a `figure` element that contains a caption. You can see much more about `:has()` in [this article by Jhey Tompkins](__TODO__).
+
+{% Aside 'caution' %}
+Current [browser support](https://caniuse.com/css-has) for `:has()` includes Chromium and Safari.
+{% endAside %}
+
+## Container queries + :has()
+
+You can combine the parent selection powers of `:has()` with the parent querying powers of container queries to create some really dynamic intrinsic styles.
+
+Let’s expand on the first example with the rocket card. What if we had a card without an image? Maybe we’d want to increase the size of the title and adjust the grid layout to single column so that it looks more intentional without the image.
```

Comment:
```suggestion
Let’s expand on the first example with the rocket card. What if you had a card without an image? Maybe you want to increase the size of the title and adjust the grid layout to single column so that it looks more intentional without the image.
```